src/views/Main/Header/LoginButton/LoginButton.js

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import classNames from 'classnames';
+import {
+  Tab,
+  Tabs,
+  Overlay,
+  Classes,
+  Button,
+} from '@blueprintjs/core';
+import { LoginForm, SignupForm } from '@terralego/core/modules/Auth';
+import withDeviceSize from '@terralego/core/hoc/withDeviceSize';
+
+import NavBarItemDesktop from '../NavBarItemDesktop';
+import NavBarItemTablet from '../NavBarItemTablet';
+
+export class LoginButton extends React.Component {
+  state = {
+    isOpen: false,
+    hasCreated: false,
+    isNewlyAuthenticated: false,
+  }
+
+  static defaultProps = {
+    isMobileSized: false,
+  }
+
+  componentDidUpdate ({ authenticated: prevAuthenticated }) {
+    const { authenticated } = this.props;
+    if (authenticated && !prevAuthenticated) {
+      this.close();
+    }
+  }
+
+  open = () => this.setState({ isOpen: true });
+
+  close = () => this.setState({ isOpen: false })
```

Comment:
Instead of both functions, maybe you could create an unique function like 
`togglePopin = isOpen => this.setState({ isOpen })`

and when you call it
` <NavBarItem {...this.props} onClick={() => this.togglePopin(true)} />`